### ISSUE TITLE
Support for AWS S3 redirection rules.

### ISSUE THREAD

**Issue by aldarund** [hashicorp/terraform#4907]
S3 allows to specify a custom redirection rules.
Here docs : http://docs.aws.amazon.com/AmazonS3/latest/dev/HowDoIWebsiteConfiguration.html
For example:
 ```
<RoutingRules>
    <RoutingRule>
    <Condition>
       <KeyPrefixEquals>images/</KeyPrefixEquals>
    </Condition>
    <Redirect>
      <ReplaceKeyWith>folderdeleted.html</ReplaceKeyWith>
    </Redirect>
    </RoutingRule>
  </RoutingRules>
```
But looks like its not possible to set such rules using terraform.

**Comment by catsby** [comment]
Thanks @aldarund – doesn't look like we have this yet, but the API docs are here for anyone who happens to get to it before I do!

- http://docs.aws.amazon.com/fr_fr/AmazonS3/latest/API/RESTBucketPUTwebsite.html

**Comment by optimisticanshul** [comment]
@catsby if no one started this yet, i can try to implement this :+1:

**Comment by catsby** [comment]
Thanks @optimisticanshul – I am unaware of anyone working on this at the moment

**Comment by stack72** [comment]
Hi @aldarund 

this was merged in #5327

Paul
